### PR TITLE
Tsifications galore

### PR DESCRIPTION
**Pull request by OfekShilon**
Still mostly mindless numb work, one rename that touched many files (`getArgumentParser` -> `getArgumentParserCls`), and a few real improvements. 

The goal is to be able to turn on `noImplicitAny` for the project, to enforce higher code quality. 600  violations to go.

My old suggestion of separating similar work (in the future) to many small PRs stands, if it would make reviewing easier.

### CODE REVIEW

File: lib/compilers/assembly.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ export class AssemblyCompiler extends BaseCompiler {
         return [];
     }
 
-    override getArgumentParser() {
+    override getArgumentParserCls() {
```
Comment (mattgodbolt) [inline lib/compilers/assembly.ts:54]: What triggered this rename? And why "Cls" and not "Class"?

**Comment by mattgodbolt** [comment]
Thanks Ofek; a skim-over and it looked ok, just some "why not call it Class not Cls" ? And...why was the rename needed? mostly curiosity there.

BTW if you create the branches in the CE project itself (instead of in your fork), I can check it out easier and then e.g. resolve conflicts for you) Up to you though :)

File: lib/compilers/assembly.ts (hunk 1 above)
Comment (OfekShilon) [reply lib/compilers/assembly.ts:54]: This doesn't actually return a parser, but the *class* object of a parser. Later on class methods are invoked on the returned object with '.' syntax.  That was very confusing, and only after realizing it was I able to fix the return hint from any to a proper type.

I'm fine with renaming to *Class if you prefer.

**Comment by OfekShilon** [comment]
> BTW if you create the branches in the CE project itself (instead of in your fork), I can check it out easier and then e.g. resolve conflicts for you) Up to you though :)

Good point. Will do in the future.

The rename thing (addressed above) is an example for stuff that could be easily separated to small standalone pr. Do you prefer to review many small prs over a single mixed one?

File: lib/compilers/assembly.ts (hunk 1 above)
Comment (OfekShilon) [reply lib/compilers/assembly.ts]: Yes I see now that *Class is the CE naming convention. Pushed a re-rename.

**Comment by OfekShilon** [comment]
I think I'll make future tsifications in many small PRs. Not only will these be more reviewable, rebasing before push would be less painful.

**Pull request merged**